BridgeCareApp/VuejsApp/src/services/scenario.service.ts

```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
         });
     }
 
-    static getLegacyScenarios(scenarios: Scenario[]): AxiosPromise {
+    /**
+     * Given two arrays of scenarios, returns an array of
+     * scenarios that are in the first array but not the second.
+     * @param arrayA The first array 
+     * @param arrayB The second array
+     */
+    private static getMissingScenarios(arrayA: Scenario[], arrayB: Scenario[]): Scenario[]  {
+        var missingScenarios: Scenario[] = [];
+            arrayA.forEach(simulation => {
+                if (!any(propEq('simulationId', simulation.simulationId), arrayB)) {
+                    missingScenarios.push(simulation);
+                }
+            });
+            return missingScenarios;
+    }
+
+    /**
+     * Removes several scenarios from mongoDB
+     * @param scenarios Array of scenarios to remove
+     */
+    private static removeMongoScenarios(scenarios: Scenario[]): AxiosPromise {
         return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
-            axiosInstance.get('api/GetScenarios')
-                .then((responseLegacy: AxiosResponse<Scenario[]>) => {
-                    if (hasValue(responseLegacy)) {
-                        var resultant: Scenario[] = [];
-                        responseLegacy.data.forEach(simulation => {
-                            if (!any(propEq('simulationId', simulation.simulationId), scenarios)) {
-                                simulation.shared = false;
-                                simulation.status = 'N/A';
-                                resultant.push(simulation);
-                            }
-                        });
+            scenarios.forEach(simulation => {
+                nodejsAxiosInstance.delete(`api/DeleteMongoScenario/${(simulation as any)._id}`)
+                    .then((res: AxiosResponse) => {
+                        if (hasValue(res)) {
+                            return resolve(res);
+                        }
+                    })
+                    .catch((error: any) => {
+                        return resolve(error.response);
+                    });
+            });
+        });
+    }
 
-                        if (resultant.length != 0) {
-                            nodejsAxiosInstance.post('api/AddMultipleScenarios', resultant)
-                                .then((res: AxiosResponse<Scenario[]>) => {
+    /**
+     * Updates mongodb to match the scenarios from the legacy database
+     */
+    static getLegacyScenarios(): AxiosPromise {
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+            axiosInstance.get('api/GetScenarios')
+                .then((responseLegacy: AxiosResponse<Scenario[]>)=>{
+                    if (hasValue(responseLegacy)){
+                    this.getMongoScenarios()
+                        .then((responseMongo: AxiosResponse<Scenario[]>)=>{
+                            if (hasValue(responseMongo)) {
```

Comment:
Responses from mongo aren't as cut & dry as responses from the C# application. I assumed you tested, but if you look at other Vue services there is a required conversion from mongo's data structure to the one used by Vue.

Reply:
That conversion takes place in the caller of this function (the getLegacyScenarios action from scenario.module.ts).

Should I move the conversion into the service, or leave it as part of the action?

Reply:
I didn't notice that function call. This is fine.